bindings/gumjs/gumquickprocess.c

```diff
@@ -622,3 +635,61 @@ gum_quick_exception_handler_on_exception (GumExceptionDetails * details,
 
   return handled;
 }
+
+GUMJS_DEFINE_FUNCTION (gumjs_process_run_on_thread)
+{
+  GumQuickScope scope = GUM_QUICK_SCOPE_INIT (core);
+  GumThreadId thread_id;
+  JSValue user_func;
+  GumQuickRunOnThreadContext sync_ctx;
+  GumStalker * stalker;
+  gboolean success;
+
+  if (!_gum_quick_args_parse (args, "ZF", &thread_id, &user_func))
+    return JS_EXCEPTION;
+
+  if (thread_id == 0)
+    return JS_UNDEFINED;
+
+  _gum_quick_scope_suspend (&scope);
+
+  sync_ctx.core = core;
+  sync_ctx.scope = scope;
+  sync_ctx.user_func = JS_DupValue (core->ctx, user_func);
+  sync_ctx.sync = FALSE;
+
+  stalker = gum_stalker_new ();
+
+  success = gum_stalker_run_on_thread (stalker, thread_id,
+      gum_js_process_run_cb, &sync_ctx);
+  _gum_quick_scope_resume (&scope);
+
+  while (gum_stalker_garbage_collect (stalker))
+    g_usleep (10000);
+
+  g_object_unref (stalker);
+
+  if (success)
+  {
+    return JS_UNDEFINED;
+  }
+  else
+  {
+    _gum_quick_throw_literal (ctx, "Failed to run on thread");
+    return JS_EXCEPTION;
+  }
+}
+
+static void
+gum_js_process_run_cb (const GumCpuContext * cpu_context,
+                       gpointer user_data)
+{
+  GumQuickRunOnThreadContext * sync_ctx =
+      (GumQuickRunOnThreadContext *) user_data;
```

Comment:
No need to cast from `void *` in C (C++ requires it, though).

Reply:
Fixed.